### PR TITLE
[TT-7683] OAS plugin bundle

### PR DESCRIPTION
**Pull request by jeffy-mathew**
jira - https://tyktech.atlassian.net/browse/TT-7683

### CODE REVIEW

**Review by titpetric — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4675/merge
Commit: 359da0d58c80900f706e5c34d8ca50ca13ed470d 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3986750603)

File: apidef/oas/middleware_test.go [hunk 1]
```diff
@@ -68,6 +68,27 @@ func TestPluginConfig(t *testing.T) {
 			assert.Equal(t, pluginConfig, newPluginConfig)
 		}
 	})
+
+	t.Run("bundle", func(t *testing.T) {
```
Comment (furkansenharputlu) [inline apidef/oas/middleware_test.go:72]: Can you please add an empty case check?

File: apidef/oas/middleware.go [hunk 2]
```diff
@@ -108,16 +108,65 @@ type PluginConfig struct {
...
+	// Tyk classic API definition: `custom_middleware_bundle_disabled`
+	Enabled bool `bson:"enabled" json:"enabled"` // required.
+	// Path is the path suffix to construct the URL to fetch plugin bundle from.
+	// Path will be suffixed to `bundle_base_url` in gateway config.
+	Path string `bson:"path" json:"path"` // required.
+}
+
+// Fill fills PluginBundle from apidef.
+func (p *PluginBundle) Fill(api apidef.APIDefinition) {
+	if !api.CustomMiddlewareBundleDisabled && api.CustomMiddlewareBundle == "" {
```
Comment (furkansenharputlu) [inline apidef/oas/middleware.go:152]: Shouldn't this be `api.CustomMiddlewareBundleDisabled? && api.CustomMiddlewareBundle == ""`?

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4675/merge
Commit: 1821da20dff32b9123a2a57cb6909b55eae0fc06 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3986903864)

File: apidef/oas/middleware.go (hunk 2 above)
Comment (jeffy-mathew) [reply apidef/oas/middleware.go:152]: we're checking for the default values here, so `!api.CustomMiddlewareBundleDisabled && api.CustomMiddlewareBundle == "" ` is the expected condition right?

File: apidef/oas/middleware_test.go (hunk 1 above)
Comment (jeffy-mathew) [reply apidef/oas/middleware_test.go:72]: https://github.com/TykTechnologies/tyk/pull/4675/files#diff-0af31cb29ae298a6ac3e402b283ab364a6fd793fd04f253ef7c4983234c17befR36 - empty case is added here

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4675/merge
Commit: 15830ba2e0ec76e58513663a3d07e0cfa585833a 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3987138811)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4675/merge
Commit: a770717e1267a825d6f09e175d7b8c0af97fe531 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3987621042)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4675/merge
Commit: 09666a2c5cf1b3d4b4473e836fa08c04a73e330c 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3993088895)

**Review by furkansenharputlu — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4675/merge
Commit: f3b57107a6dfd3bb522c94fdfdee3e8b1be94cc2 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3993629463)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4675/merge
Commit: f32ba94af12fe7a6afb415adc67d639a9bd2a23d 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3993644473)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4675/merge
Commit: 57b46d559213d2cb8a8282aebbf2bb1040c308bb 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3993674120)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4675/merge
Commit: aafbd01b87ec642434454ed6b5de99383b246f11 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3995852097)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4675/merge
Commit: dfc72c33a5f2142bfb195a455510de394d52d220 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3996117372)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4675/merge
Commit: 7258f70c5b903a36eeca2c062a64b36924bd2fbf 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3996213008)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4675/merge
Commit: 6397426a6e19cd05e1f705134650778bc243f7fb 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3996224182)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4675/merge
Commit: 086ffc4707a37b01a056301fa322f799f0ab6c06 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3996344028)

**Pull request merged**